From b90df318ef0078bf4d77778b8fa235cdd1197848 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Wed, 4 Jan 2017 15:36:00 +0800
Subject: [PATCH] NVMe: tage the disk with flag of is_rk_disk

For rockchip platform, we may need to support
NVMe bootup, so we need to add this flag so that
the rk partition layout could be able to find the
correct partition from parameter.

Change-Id: I3e0213df893bd137fa6d5fd0a0120cabeb6259e8
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/nvme/host/pci.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/nvme/host/pci.c b/drivers/nvme/host/pci.c
index 8bb2c666eb4c..37c5461050d0 100644
--- a/drivers/nvme/host/pci.c
+++ b/drivers/nvme/host/pci.c
@@ -2290,6 +2290,11 @@ static void nvme_alloc_ns(struct nvme_dev *dev, unsigned nsid)
 	disk->queue = ns->queue;
 	disk->driverfs_dev = dev->device;
 	disk->flags = GENHD_FL_EXT_DEVT;
+#ifdef CONFIG_ARCH_ROCKCHIP
+	disk->is_rk_disk = true;
+#else
+	disk->is_rk_disk = false;
+#endif
 	sprintf(disk->disk_name, "nvme%dn%d", dev->instance, nsid);
 
 	/*
-- 
2.35.3

